symphony/lib/toolkit/class.general.php

```diff
@@ -312,46 +312,53 @@ public static function createHandle($string, $max_length = 255, $delim = '-', $u
     {
         $max_length = intval($max_length);
 
-        // Strip out any tag
+        // make it lowercase
+
+        $string = strtolower($string);
+
+        // strip out tags
+
         $string = strip_tags($string);
 
-        // Remove punctuation
-        $string = preg_replace('/[\\.\'"]+/', null, $string);
+        // remove unwanted characters
+
+        $string = preg_replace('/[^\w- ]+/', null, $string);
```

Comment:
> This new createHandle returns dog-cat32lf-dfajl without this change.

You mean invalid characters should better be replaced by `-` instead of just being stripped out?